tests/unit/include/bout/test_stencil.cxx

```diff
@@ -0,0 +1,272 @@
+    auto offset = this->noOffset.xm(i);
+    EXPECT_EQ(offset.dx, -i);
+  }
+}
+
+TYPED_TEST(IndexOffsetStructTests, YPlus) {
+  auto offset = this->noOffset.yp();
+  EXPECT_EQ(offset.dy, 1);
+  for (int i = 0; i < 5; i++) {
+    auto offset = this->noOffset.yp(i);
+    EXPECT_EQ(offset.dy, i);
+  }
+}
+
+TYPED_TEST(IndexOffsetStructTests, YMinus) {
+  auto offset = this->noOffset.ym();
+  EXPECT_EQ(offset.dy, -1);
+  for (int i = 0; i < 5; i++) {
+    auto offset = this->noOffset.ym(i);
+    EXPECT_EQ(offset.dy, -i);
+  }
+}
+
+TYPED_TEST(IndexOffsetStructTests, ZPlus) {
+  auto offset = this->noOffset.zp();
+  EXPECT_EQ(offset.dz, 1);
+  for (int i = 0; i < 5; i++) {
+    auto offset = this->noOffset.zp(i);
+    EXPECT_EQ(offset.dz, i);
+  }
+}
+
+TYPED_TEST(IndexOffsetStructTests, ZMinus) {
+  auto offset = this->noOffset.zm();
+  EXPECT_EQ(offset.dz, -1);
+  for (int i = 0; i < 5; i++) {
+    auto offset = this->noOffset.zm(i);
+    EXPECT_EQ(offset.dz, -i);
+  }
+}
+
+TYPED_TEST(IndexOffsetStructTests, Add) {
+  IndexOffset<TypeParam> offset1 = {1, 1, 1}, offset2 = {1, 2, 1}, sum = {2, 3, 2};
+
+  EXPECT_EQ(this->noOffset + offset1, offset1);
+  EXPECT_EQ(offset1 + offset2, sum);
+}
+
+TYPED_TEST(IndexOffsetStructTests, Subtract) {
+  IndexOffset<TypeParam> offset1 = {1, 1, 1}, offset2 = {1, 2, 1}, difference = {0, 1, 0};
+  EXPECT_EQ(offset1 - this->noOffset, offset1);
+  EXPECT_EQ(offset2 - offset1, difference);
+}
+
+TYPED_TEST(IndexOffsetStructTests, AddToIndex) {
+  IndexOffset<TypeParam> offset1 = {1, 0, 0}, offset2 = {0, 2, 0}, offset3 = {0, 0, 11},
+                         offset4 = {2, 3, -2};
+  EXPECT_EQ(this->zero + offset1, this->zero.xp());
+  EXPECT_EQ(offset1 + this->zero, this->zero.xp());
+  if (!std::is_same<TypeParam, IndPerp>::value) {
+    EXPECT_EQ(this->zero + offset2, this->zero.yp(2));
+    EXPECT_EQ(offset2 + this->zero, this->zero.yp(2));
+  }
+  if (!std::is_same<TypeParam, Ind2D>::value) {
+    EXPECT_EQ(this->zero + offset3, this->zero.zp(11));
+    EXPECT_EQ(offset3 + this->zero, this->zero.zp(11));
+  }
+  if (std::is_same<TypeParam, Ind3D>::value) {
+    EXPECT_EQ(this->zero + offset4, this->zero.xp(2).yp(3).zm(2));
+    EXPECT_EQ(offset4 + this->zero, this->zero.xp(2).yp(3).zm(2));
+  }
+}
+
+TYPED_TEST(IndexOffsetStructTests, SubtractFromIndex) {
+  IndexOffset<TypeParam> offset1 = {1, 0, 0}, offset2 = {0, 2, 0}, offset3 = {0, 0, 11},
+                         offset4 = {2, 3, -2};
+  EXPECT_EQ(this->zero - offset1, this->zero.xm());
+  if (!std::is_same<TypeParam, IndPerp>::value) {
+    EXPECT_EQ(this->zero - offset2, this->zero.ym(2));
+  }
+  if (!std::is_same<TypeParam, Ind2D>::value) {
+    EXPECT_EQ(this->zero - offset3, this->zero.zm(11));
+  }
+  if (std::is_same<TypeParam, Ind3D>::value) {
+    EXPECT_EQ(this->zero - offset4, this->zero.xm(2).ym(3).zp(2));
+  }
+}
+
+
+template <typename T>
+class StencilUnitTests :  public ::testing::Test {
+public:
+  //WithQuietOutput all{output};
```

Comment:
Commented out code

Reply:
This should be uncommented. I must have commented it out during debugging and forgotten about it.